src/Microsoft.TestPlatform.Client/Discovery/DiscoveryRequest.cs

```diff
@@ -248,6 +250,22 @@ public void HandleDiscoveryComplete(DiscoveryCompleteEventArgs discoveryComplete
                     OnDiscoveredTests.SafeInvoke(this, discoveredTestsEvent, "DiscoveryRequest.DiscoveryComplete");
                 }
 
+                // Add extensions discovered by vstest.console.
+                //
+                // TODO(copoiena): Writing telemetry twice is less than ideal.
+                // We first write telemetry data in the _requestData variable in the ParallelRunEventsHandler
+                // and then we write again here. We should refactor this code and write only once.
```

Comment:
The first time, we write the telemetry data in the `_requestData` variable that is used only for logging/debugging purposes (i.e. so that we can check the files created on the local machine and make sure they contain our telemetry keys).

Then we write *here* the actual telemetry that is passed to the TranslationLayer in order to be handled by VS.

There's and identical set of comments for test run, same goes for that operation.